### PR TITLE
Add unit test for new task

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Summary
Add unit test for new task structure for object detection.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/detection/adapters/mmdet/task.py [hunk 1]
```diff
@@ -339,8 +339,8 @@ def _infer_model(
         mm_dataset = build_dataset(cfg.data.test)
         dataloader = build_dataloader(
             mm_dataset,
-            samples_per_gpu=cfg.data.get("samples_per_gpu", 1),
-            workers_per_gpu=cfg.data.get("workers_per_gpu", 0),
+            samples_per_gpu=cfg.data.test_dataloader.get("samples_per_gpu", 1),
```
Comment (eunwoosh) [inline otx/algorithms/detection/adapters/mmdet/task.py]:
```suggestion
            samples_per_gpu=samples_per_gpu,
```

File: tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def setup(self, mocker):
+        def mock_init_logger():
+            pass
+
+        def mock_configure(model_cfg, model_ckpt, data_cfg, training=False, **kwargs):
+            return mmcv.ConfigDict()
+
+        self.exporter = Exporter()
+        self.exporter._init_logger = mock_init_logger
+        # self.exporter.configure = mock_configure
```
Comment (eunwoosh) [inline tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py]: Is these comments necessary? If not, how about removing them?

File: tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        # self.exporter.mode = ["mock_mode", "train"]
```
Comment (eunwoosh) [inline tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py]: Same as above

File: tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        self.fake_config = mmcv.ConfigDict(work_dir="/path/work_dir", data=dict(test=dict(dataset=mocker.MagicMock())))
+        # mocker.patch.object(self.exporter, "configure", return_value=fake_config)
```
Comment (eunwoosh) [inline tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py]: Same as above

File: tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import mmcv
+import pytest
+
+from otx.algorithms.common.adapters.mmcv.tasks.exporter import Exporter
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class TestExporterMixin:
```
Comment (eunwoosh) [inline tests/unit/algorithms/common/adapters/mmcv/tasks/test_exporter.py]:
```suggestion
class TestExporter:
```

**Review by harimkang — DISMISSED** [review]

File: tests/unit/algorithms/detection/adapters/mmdet/hooks/test_det_saliency_map_hook.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                type="SSDAnchorGenerator",
+                basesize_ratio_range=(0.15, 0.9),
+                strides=(16, 32, 48),
+                ratios=[[0.5], [0.1], [0.3]],
+            ),
+            act_cfg={},
+        )
+        self.hook = DetSaliencyMapHook(self.module)
+        assert self.hook._get_cls_scores_from_feature_map(torch.Tensor(1, 3, 512, 32, 32)) is not None
+        self.module.bbox_head = torch.nn.Module()
```
Comment (eunwoosh) [inline tests/unit/algorithms/detection/adapters/mmdet/hooks/test_det_saliency_map_hook.py]: I think it's better to separate code from this line to below as a dependent test case.

File: tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            paired_bn=True,
+        )
+
+    @e2e_pytest_unit
+    def test_init_weights(self):
+        self.model.init_weights()
+
+    @e2e_pytest_unit
+    def test_forward(self):
+        assert self.model.inputs == self.model._inputs
```
Comment (eunwoosh) [inline tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py]: Is this part necessary? I think checking not public attribute is weird.

File: tests/unit/algorithms/detection/adapters/mmdet/models/roi_heads/roi_extractors/test_ov_mmdet_single_level_roi_extractor.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    def test_build_roi_layers(self):
+        extractor = SingleRoIExtractor(
+            roi_layer=dict(type="RoIInterpolationPool", output_size=14, mode="bilinear"),
+            out_channels=1024,
+            featmap_strides=[8],
+        )
+        assert all(isinstance(layer, RoIInterpolationPool) for layer in extractor.roi_layers)
+
+
+#  class TestRoIInterpolationPool:
```
Comment (eunwoosh) [inline tests/unit/algorithms/detection/adapters/mmdet/models/roi_heads/roi_extractors/test_ov_mmdet_single_level_roi_extractor.py]: How about removing these comments?

File: tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+from tests.unit.algorithms.common.adapters.mmcv.nncf.test_helpers import (
+    create_config,
+    create_dataset,
+    create_model,
+)
+
+
+@e2e_pytest_unit
+def test_build_nncf_detector():
```
Comment (eunwoosh) [inline tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: I think this test function can be separated by each dependent test case (maybe 3?).
How about separating to keep one test for one case?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left some comments. please take a look.

File: tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py (hunk 7 above)
Comment (jaegukhyun) [reply tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py]: Actually, this is not mine. I deleted this file accidently, so I just restore this

File: tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py (hunk 9 above)
Comment (jaegukhyun) [reply tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: This is not mine too, I just restored what I accidently deleted

File: tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py (hunk 7 above)
Comment (eunwoosh) [reply tests/unit/algorithms/detection/adapters/mmdet/models/backones/test_ov_mmdet_mmov_backbone.py]: I think this code doesn't mean more than checking there are property functions to each attribute. How about just removing these lines?

File: tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py (hunk 9 above)
Comment (eunwoosh) [reply tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: I see. but how about separating them at this time to make it more readable?

Comment (jaegukhyun) [reply tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: I'm not sure, This function is test for one function "build_nncf_detector" and it just check some various situations. I don't want to refactor this whole test in this PR.

Comment (eunwoosh) [reply tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: Yes you're right. Refactoring whole test cases is impossible. I just want to make test function a little clean step by step. If you don't have time for this, then I can do this. Could I add commit for this?

Comment (jaegukhyun) [reply tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py]: Sure, why not?

File: tests/unit/algorithms/detection/adapters/mmdet/hooks/test_det_saliency_map_hook.py (hunk 6 above)
Comment (eunwoosh) [reply tests/unit/algorithms/detection/adapters/mmdet/hooks/test_det_saliency_map_hook.py]: I'll modify it with my commit

**Review by eunwoosh — DISMISSED** [review]
Thanks for your work.
I modified files below:
tests/unit/algorithms/detection/adapters/mmdet/hooks/test_det_class_probability_map_hook.py
tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py
If you have some comments to that, please let me know :)

**Comment by jaegukhyun** [comment]
@eunwoosh Could you solve pre-commit issue?

**Review by eunwoosh — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Pull request merged**